File: frontend/src/types/job.ts
export interface Job {
  _id: string;
  externalId: string;
  title: string;
  company: string;
  location: string;
  description: string;
  url: string;
  source: string;
  postedAt: string;
  scrapedAt: string;
  remote: boolean;
  tags: string[];
  active: boolean;
}

export interface JobResponse {
  jobs: Job[];
  total: number;
  currentPage: number;
  totalPages: number;
}

File: scrapers/internal/models/job.go
package models

import "time"

// Job represents a standardized job listing
type Job struct {
	ID          string    `json:"externalId"`
	Title       string    `json:"title"`
	Company     string    `json:"company"`
	Location    string    `json:"location"`
	Description string    `json:"description"`
	URL         string    `json:"url"`
	Source      string    `json:"source"` // e.g., "LinkedIn", "Indeed"
	PostedAt    time.Time `json:"postedAt"`
	ScrapedAt   time.Time `json:"scrapedAt"`
	Remote      bool      `json:"remote"`
	Salary      string    `json:"salary,omitempty"`
	Tags        []string  `json:"tags,omitempty"`
}
